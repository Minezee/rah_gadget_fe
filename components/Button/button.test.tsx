import {
  cleanup,
  fireEvent,
  render,
  screen,
  waitFor,
} from "@testing-library/react";
import "@testing-library/jest-dom";
import TestRenderer from "react-test-renderer";
import { ButtonArrow } from "./button";

describe("Button arrow", () => {
  test("testing box", async () => {
    render(<ButtonArrow>Lihat Semua Laptop</ButtonArrow>);

    const buttonTest = screen.getByRole("button", {
      name: /Lihat Semua Laptop/i,
    });
    expect(buttonTest).toBeInTheDocument();

    const buttonArrowTest = screen.getByTestId("button-arrow");
    expect(buttonArrowTest).not.toBeVisible()

    // Cari cara untuk test show component when hover button
  });

  test("Matching snapshots", () => {
    const tree = TestRenderer.create(
      <ButtonArrow>Lihat Semua Laptop</ButtonArrow>
    );

    expect(tree).toMatchSnapshot()
  });
});
