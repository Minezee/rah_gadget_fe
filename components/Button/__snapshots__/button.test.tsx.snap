// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button arrow Matching snapshots 1`] = `
<button
  className="sc-bkkeKt sc-ieecCq kNRUrJ dbnMPG"
>
  <h1
    className="sc-furwcr cyDOeD"
  >
    Lihat Semua Laptop
  </h1>
  <svg
    data-testid="button-arrow"
    fill="none"
    height="24"
    style={
      Object {
        "marginLeft": "16px",
      }
    }
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M5 12H19"
      stroke="white"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="3"
    />
    <path
      d="M12 5L19 12L12 19"
      stroke="white"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="3"
    />
  </svg>
</button>
`;
